### PR TITLE
#176 Fixes corrupted media file downloads

### PR DESCRIPTION
**Pull request by allishultes**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      

#176 

**Describe what the PR does**    

- [x] Passes file type to the downloadjs function, which fixed media files downloading as uris
- [x] Opens a stream to ensure the whole media file is loaded before being saved locally (fixes the corrupted files warnings)
- [x] Lightly refactors the MediaModal component to give feedback on download progress

Screenshot 2021-04-08 at 18 39 37

**State whether the PR is ready for review or whether it needs extra work**    

Ready for review 

**Additional context**    

To test:

1. Navigate to a PaperEdit
2. Click 'Export'
3. Try to export all the files and then open them and confirm you aren't getting any corrupted file warnings, etc

I've validated this on my end but please check it's fixed with any corrupted files you've noticed!

### CODE REVIEW

File: src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js [hunk 1]
```diff
@@ -1,16 +1,60 @@
...
+        size="sm"
+        disabled={ isDownloading }
+        onClick={ () => handleClick(url, fileName, type) }>
+        <FontAwesomeIcon icon={ faDownload } />
+      </Button>
+      <p className={ styles.mediaName }>{name}</p>
+      {getStatus()}
+    </li>
+  );
 };
```
Comment (emettely) [inline src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js:58]: Should be its own file

File: src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js [hunk 2]
```diff
@@ -1,16 +1,60 @@
...
-import React from 'react';
+import React, { useState } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import {
   faDownload
 } from '@fortawesome/free-solid-svg-icons';
 
 import Modal from 'react-bootstrap/Modal';
 import Button from 'react-bootstrap/Button';
+import styles from './index.module.css';
```
Comment (emettely) [inline src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js:10]: This is just a general preference for this project - the styling should be in-line where possible!

File: src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js [hunk 3]
```diff
@@ -1,16 +1,60 @@
...
+  const [ progress, setProgress ] = useState();
+
+  const handleProgress = (e) => {
+    if (e.lengthComputable) {
+      setProgress(Math.round((e.loaded / e.total) * 100));
+    }
+  };
+
+  const handleClick = () => {
+    const x = new XMLHttpRequest();
```
Comment (emettely) [inline src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js:25]: there might be a better name than x :)

File: src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js [hunk 4]
```diff
@@ -1,16 +1,60 @@
...
+      download(x.response, fileName, type);
+      setIsDownloading(false);
+    };
+    x.send();
+  };
+
+  const getStatus = () => {
+    if (isDownloading) {
+      return (
+        <span className={ styles.status }>Downloading... { progress }%</span>
```
Comment (emettely) [inline src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js:40]: as mentioned above, would prefer for it to look like:
```suggestion
        <span style={{ css thing }}>Downloading... { progress }%</span>
```

File: src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js [hunk 5]
```diff
@@ -1,16 +1,60 @@
...
+      );
+    }
+  };
+
+  return (
+    <li key={ name } className={ styles.mediaItem }>
```
Comment (emettely) [inline src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js:46]: same as above

File: src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js [hunk 6]
```diff
@@ -1,16 +1,60 @@
...
+      <Button
+        variant="outline-primary"
+        size="sm"
+        disabled={ isDownloading }
+        onClick={ () => handleClick(url, fileName, type) }>
+        <FontAwesomeIcon icon={ faDownload } />
+      </Button>
+      <p className={ styles.mediaName }>{name}</p>
```
Comment (emettely) [inline src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js:54]: and here :)

File: src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js [hunk 7]
```diff
@@ -1,16 +1,60 @@
...
 import download from 'downloadjs';
 
-const handleClick = (url, fileName) => {
-  download(url, fileName, 'application/octet-stream');
+const DownloadItem = ({ name, fileName, url, type }) => {
+  const [ isDownloading, setIsDownloading ] = useState(false);
+  const [ progress, setProgress ] = useState();
+
+  const handleProgress = (e) => {
+    if (e.lengthComputable) {
```
Comment (emettely) [inline src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/ExportModal/MediaModal.js:19]: oooh

**Review by emettely — APPROVED** [review]
looks great! Just one thing about CSS styles, which is a non-issue.

**Comment by emettely** [comment]
Screenshot 2021-04-09 at 15 45 21
Titles that are too long can be a problem, but not that important.

**Review by sarahrainbow — APPROVED** [review]
looks great 💫

**Pull request merged**